client/new-stack-editor/lib/views/wizard/providerselectionview.coffee

```diff
@@ -0,0 +1,67 @@
+kd = require 'kd'
+JView = require 'app/jview'
+globals = require 'globals'
+Tracker = require 'app/util/tracker'
+
+
+module.exports = class ProviderSelectionView extends JView
+
+  constructor: (options = {}, data) ->
+
+    options.cssClass ?= 'provider-selection'
+
+    super options, data
+
+    @createProviders()
+
+
+  createProviders: ->
+
+    { providers } = globals.config
+
+    supportedProviders = (Object.keys providers).filter (p) ->
+      return p  if providers[p].supported
+
+    @providers = new kd.CustomHTMLView
+      cssClass: 'providers box-wrapper clearfix'
+
+    supportedProviders.forEach (provider) =>
+
+      _provider = providers[provider]
+
+      if not _provider.enabled
+        extraClass = 'coming-soon'
+        stateLabel = 'Coming soon'
+      else if _provider.enabled is 'beta'
+        extraClass = 'beta'
+        stateLabel = 'BETA'
+      else
+        extraClass = ''
+        stateLabel = ''
+
+      @providers.addSubView providerView = new kd.CustomHTMLView
+        cssClass   : "provider box #{provider} #{extraClass}"
+        provider   : provider
+        attributes : { 'data-before-content': stateLabel }
+        click      : =>
+          return  if extraClass is 'coming-soon'
+
+          Tracker.track Tracker["STACKS_WIZARD_SELECTED_#{provider.toUpperCase()}"]
+
+          providerView.setClass 'selected'
+          @selected?.unsetClass 'selected'
+
+          @selected = if @selected is providerView then null else providerView
+          @emit 'SelectedProviderChanged', @selected
```

Comment:
Recently i have started to extract out this kind of hardcoded values into a `constants.coffee` right next to the files that uses them. It's especially useful when one wants to change the value of the event, change in one place, and don't think about where it was being used.